tst/EnergyPlus/unit/HVACDXSystem.unit.cc

```diff
@@ -801,4 +803,294 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_LatentDegradation_Test)
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    std::string compName = "DX COOLING COIL SYSTEM";
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->MinutesPerTimeStep = 60;
+    state->dataZoneEquip->ZoneEquipInputsFilled = true;
+    state->dataGlobal->NumOfZones = 1;
+    state->dataZoneEquip->ZoneEquipConfig.allocate(1);
+    state->dataZoneEquip->ZoneEquipConfig(1).NumExhaustNodes = 1;
+    state->dataZoneEquip->ZoneEquipConfig(1).InletNode.allocate(1);
+    state->dataZoneEquip->ZoneEquipConfig(1).ExhaustNode.allocate(1);
+
+    state->dataHVACGlobal->NumPrimaryAirSys = 1;
+    state->dataAirSystemsData->PrimaryAirSystems.allocate(1);
+    state->dataAirLoop->AirLoopControlInfo.allocate(1);
+    state->dataAirSystemsData->PrimaryAirSystems(1).NumBranches = 1;
+    state->dataAirSystemsData->PrimaryAirSystems(1).Branch.allocate(1);
+    state->dataAirSystemsData->PrimaryAirSystems(1).Branch(1).TotalComponents = 1;
+    state->dataAirSystemsData->PrimaryAirSystems(1).Branch(1).Comp.allocate(1);
+    state->dataAirSystemsData->PrimaryAirSystems(1).Branch(1).Comp(1).Name = compName;
+    OutputReportPredefined::SetPredefinedTables(*state);
+    ScheduleManager::ProcessScheduleInput(*state);
+    state->dataScheduleMgr->Schedule(1).CurrentValue = 1.0; // Enable schedule without calling schedule manager
+
+    bool FirstHVACIteration = true;
+    bool HXUnitOn = false;
+    int InletNode = 1;
+    int ControlNode = 2; // same as outlet node number
+    int condenserNode = 3;
+    int airLoopNum = 1;
+
+    bool zoneEquipment = true;
+    UnitarySystems::UnitarySys::factory(*state, DataHVACGlobals::UnitarySys_AnyCoilType, compName, zoneEquipment, 0);
+    UnitarySystems::UnitarySys *thisSys = &state->dataUnitarySystems->unitarySys[0];
+    // call again to get the rest of the input when sysNum > -1
+    UnitarySystems::UnitarySys::getUnitarySystemInput(*state, compName, false, 0);
+
+    EXPECT_EQ(thisSys->Name, "DX COOLING COIL SYSTEM");
+    EXPECT_FALSE(thisSys->m_ISHundredPercentDOASDXCoil);
+    EXPECT_EQ(thisSys->UnitType, "CoilSystem:Cooling:DX");
+    EXPECT_EQ(thisSys->m_CoolingCoilType_Num, DataHVACGlobals::CoilDX_Cooling);
+    EXPECT_EQ(2, thisSys->m_SystemCoolControlNodeNum);
+
+    // set up outdoor environment
+    state->dataEnvrn->OutDryBulbTemp = 35.0;
+    state->dataEnvrn->OutHumRat = 0.0196;
+    state->dataEnvrn->OutBaroPress = 101325.0;
+    state->dataEnvrn->OutWetBulbTemp = 27.0932;
+    state->dataLoopNodes->Node(condenserNode).Temp = state->dataEnvrn->OutDryBulbTemp;
+    state->dataLoopNodes->Node(condenserNode).HumRat = state->dataEnvrn->OutHumRat;
+
+    // set up inputs to test coil control
+    thisSys->m_DesiredOutletTemp = 20.0;
+    thisSys->m_DesiredOutletHumRat = 1.0;
+    state->dataEnvrn->StdRhoAir = 1.2;
+    state->dataLoopNodes->Node(InletNode).MassFlowRate = 0.8 * state->dataEnvrn->StdRhoAir;
+    state->dataLoopNodes->Node(InletNode).Press = state->dataEnvrn->OutBaroPress;
```

Comment:
Failed unit test was because inlet node pressure was not defined (0) and the new test for Tsat was coming out higher than outlet node T. Adding in node Press fixed it.